File: NiukeAB/AB9/AB9.go
package main

import "fmt"

type LinkList struct {
	items []int
}

func (l *LinkList) Insert(x, y int) {
	for i := 0; i < l.Size(); i++ {
		if l.items[i] == x {
			l.items = append(l.items[:i+1], l.items[i:]...)
			l.items[i] = y
			return
		}
	}
	l.items = append(l.items, y)
}
func (l *LinkList) Size() (lgt int) {
	return len(l.items)
}
func (l *LinkList) Delete(x int) {
	for i := 0; i < l.Size(); i++ {
		if l.items[i] == x {
			l.items = append(l.items[:i], l.items[i+1:]...)
			return
		}
	}
}
func (l *LinkList) Print() {
	for i := 0; i < l.Size(); i++ {
		fmt.Print(l.items[i], " ")
	}
}
func main() {
	var ll LinkList
	var n int
	fmt.Scan(&n)
	for i := 0; i < n; i++ {
		var op string
		fmt.Scan(&op)
		switch op {
		case "insert":
			var x, y int
			fmt.Scan(&x, &y)
			ll.Insert(x, y)
		case "delete":
			var x int
			fmt.Scan(&x)
			ll.Delete(x)
		}
	}
	if ll.Size() == 0 {
		fmt.Println("NULL")
	} else {
		ll.Print()
	}
}

File: NiukeAB/AB9/AB9.py
class LinkedList():
    def __init__(self):
        self.items = []

    def insert(self, x,y):
        if x in self.items:
            idx=self.items.index(x)
            self.items.insert(idx, y)
        else:
            self.items.append(y)

    def delete(self,x):
        if x in self.items:
            idx=self.items.index(x)
            self.items.pop(idx)
    def size(self):
        return len(self.items)
    def traverse(self):
        for item in self.items:
            print(item,end=" ")
s=LinkedList()
n=int(input())
for i in range(n):
    meg=input().split(" ")
    if meg[0]=='insert':
        s.insert(meg[1],meg[2])
    elif meg[0]=='delete':
        s.delete(meg[1])
if s.size()==0:
    print("NULL")
else:
    s.traverse()
 